site/src/components/MultiSelectCombobox/MultiSelectCombobox.tsx

```diff
@@ -0,0 +1,640 @@
+>(
+	(
+		{
+			value,
+			onChange,
+			placeholder,
+			defaultOptions: arrayDefaultOptions = [],
+			options: arrayOptions,
+			delay,
+			onSearch,
+			onSearchSync,
+			loadingIndicator,
+			emptyIndicator,
+			maxSelected = Number.MAX_SAFE_INTEGER,
+			onMaxSelected,
+			hidePlaceholderWhenSelected,
+			disabled,
+			groupBy,
+			className,
+			badgeClassName,
+			selectFirstItem = true,
+			creatable = false,
+			triggerSearchOnFocus = false,
+			commandProps,
+			inputProps,
+			hideClearAllButton = false,
+		}: MultiSelectComboboxProps,
+		ref,
+	) => {
+		const inputRef = React.useRef<HTMLInputElement>(null);
+		const [open, setOpen] = React.useState(false);
+		const [onScrollbar, setOnScrollbar] = React.useState(false);
+		const [isLoading, setIsLoading] = React.useState(false);
+		const dropdownRef = React.useRef<HTMLDivElement>(null); // Added this
+
+		const [selected, setSelected] = React.useState<Option[]>(value || []);
+		const [options, setOptions] = React.useState<GroupOption>(
+			transitionToGroupOption(arrayDefaultOptions, groupBy),
+		);
+		const [inputValue, setInputValue] = React.useState("");
+		const debouncedSearchTerm = useDebouncedValue(inputValue, delay || 500);
+
+		const [previousValue, setPreviousValue] = React.useState<Option[]>(
+			value || [],
+		);
+		if (value && value !== previousValue) {
+			setPreviousValue(value);
+			setSelected(value);
+		}
+
+		React.useImperativeHandle(
+			ref,
+			() => ({
+				selectedValue: [...selected],
+				input: inputRef.current as HTMLInputElement,
+				focus: () => inputRef?.current?.focus(),
+				reset: () => setSelected([]),
+			}),
+			[selected],
+		);
+
+		const handleUnselect = React.useCallback(
+			(option: Option) => {
+				const newOptions = selected.filter((s) => s.value !== option.value);
+				setSelected(newOptions);
+				onChange?.(newOptions);
+			},
+			[onChange, selected],
+		);
+
+		const handleKeyDown = React.useCallback(
+			(e: React.KeyboardEvent<HTMLDivElement>) => {
+				const input = inputRef.current;
+				if (input) {
+					if (e.key === "Delete" || e.key === "Backspace") {
+						if (input.value === "" && selected.length > 0) {
+							const lastSelectOption = selected[selected.length - 1];
+							// If last item is fixed, we should not remove it.
+							if (!lastSelectOption.fixed) {
+								handleUnselect(selected[selected.length - 1]);
+							}
+						}
+					}
+					// This is not a default behavior of the <input /> field
+					if (e.key === "Escape") {
+						input.blur();
+					}
+				}
+			},
+			[handleUnselect, selected],
+		);
+
+		useEffect(() => {
+			const handleClickOutside = (event: MouseEvent | TouchEvent) => {
+				if (
+					dropdownRef.current &&
+					!dropdownRef.current.contains(event.target as Node) &&
+					inputRef.current &&
+					!inputRef.current.contains(event.target as Node)
+				) {
+					setOpen(false);
+					inputRef.current.blur();
+				}
+			};
+
+			if (!open) {
+				return;
+			}
```

Comment:
Could we move this to the top of the effect so that we're not creating and throwing away `handleClickOutside` when `open` is false?